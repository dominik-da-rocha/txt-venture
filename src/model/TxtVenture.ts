export interface TxtObject {}

export interface TxtPerson extends TxtObject {}

export interface TxtScene extends TxtObject {}

export interface TxtAction extends TxtObject {
  name: string;
  isEnabled: boolean;
}

export interface TxtActions extends TxtObject {
  giveTo: TxtAction;
  open: TxtAction;
  close: TxtAction;
  pickUp: TxtAction;
  lookAt: TxtAction;
  talkTo: TxtAction;
  push: TxtAction;
  pull: TxtAction;
  goto: TxtAction;
  use: TxtAction;
  save: TxtAction;
  load: TxtAction;
}

export class TxtVenture {
  title: string = "TextVenture";
  isEditor: boolean = true;
  actions: TxtActions = {
    giveTo: { name: "Gib", isEnabled: true },
    open: { name: "Öffne", isEnabled: true },
    close: { name: "Schließe", isEnabled: true },
    pickUp: { name: "Nimm", isEnabled: true },
    lookAt: { name: "Schau", isEnabled: true },
    talkTo: { name: "Rede mit", isEnabled: true },
    push: { name: "Drücke", isEnabled: true },
    pull: { name: "Ziehe", isEnabled: true },
    goto: { name: "Gehe zu", isEnabled: true },
    use: { name: "Benutze", isEnabled: true },
    save: { name: "Speichern", isEnabled: true },
    load: { name: "Laden", isEnabled: true },
  };
  objects: TxtObject[] = [];
  persons: TxtPerson[] = [];
  inventory: TxtObject[] = [];
  scenes: TxtScene[] = [];
  scene: TxtScene | null = null;
  logBook: string[] = [];

  static action(txt: TxtVenture, action: TxtAction) {
    txt.logBook.push(action.name);
  }
}

export default TxtVenture;
