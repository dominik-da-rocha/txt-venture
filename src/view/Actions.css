.Actions {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  flex-wrap: wrap;
  align-items: flex-start;
  align-content: flex-start;
}

.Actions .Action {
  display: inline;
  padding: 0.25rem;
  margin: 0.25rem;
  border: 0.1rem silver solid;
  background: white;
  border-radius: 0.25rem;
  user-select: none;
  height: 1.5rem;
}

.Actions .Action:hover {
  background: whitesmoke;
}

.Actions .Action:active {
  background: cornflowerblue;
  color: white;
  border: 0.1rem #1a1a1a solid;
}
