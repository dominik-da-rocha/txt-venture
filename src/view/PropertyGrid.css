.PropertyGrid {
  display: flex;
  flex-direction: column;
  overflow-y: auto;
  height: 40rem;
  width: 30rem;
}

.PropertyGrid > .Object .Object,
.PropertyGrid > .Object .Value {
  margin-left: 1.5rem;
}

.PropertyGrid .Object {
  display: flex;
  flex-direction: column;
}

.PropertyGrid .Value {
  display: flex;
  flex-direction: row;
  align-items: baseline;
  align-content: stretch;
  background: whitesmoke;
}

.PropertyGrid .Object > .Expander {
  display: none;
}

.PropertyGrid .Object > .Expander ~ .ExpanderContent {
  visibility: collapse;
  height: 0;
  padding: 0;
  transition: height 0.5s ease;
}

.PropertyGrid .Object > .Expander:checked ~ .ExpanderContent {
  visibility: visible;
  height: auto;
  transition: height 0.5s ease;
}

.PropertyGrid .Object > .Expander ~ .ExpanderLabel::before {
  display: inline-block;
  width: 1rem;
  content: "▶";
}

.PropertyGrid .Object > .Expander:checked ~ .ExpanderLabel::before {
  content: "▼";
}

.PropertyGrid .Object > .ExpanderLabel .PropertyGrid .Value {
  flex-direction: row;
}

.PropertyGrid .Object > .ExpanderLabel,
.PropertyGrid .Object > .ExpanderLabel::before {
  display: inline-block;
  user-select: none;
  background: whitesmoke;
}

.PropertyGrid .Object > .ExpanderLabel::before {
  width: 1rem;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  text-align: center;
}

.PropertyGrid .Object > .ExpanderLabel:hover,
.PropertyGrid .Object > .ExpanderLabel:hover::before,
.PropertyGrid .Value:hover {
  background: #e0e0e0;
}

.PropertyGrid .Object > .ExpanderLabel:active,
.PropertyGrid .Object > .ExpanderLabel:active::before,
.PropertyGrid .Value:active {
  background: cornflowerblue;
  color: white;
}

.PropertyGrid .Value > .ExpanderLabel {
  width: 30%;
  user-select: none;
}

.PropertyGrid .Value > .Editor {
  flex: 1;
  margin: 0.1rem;
}
