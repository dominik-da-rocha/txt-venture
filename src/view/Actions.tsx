import "./Actions.css";
import React from "react";
import Component, { ComponentProps, ComponentState } from "./Component";
import TxtVenture, { TxtActions, TxtAction } from "../model/TxtVenture";
import ModalDialog from "./ModalDialog";
import PropertyGrid from "./PropertyGrid";

export interface ActionsProps extends ComponentProps {
  txt: TxtVenture;
  onAction(action: TxtAction): void;
  onChange(actions: TxtActions): void;
  showEditDialog?: boolean;
}

export interface ActionsState extends ComponentState {
  showEditDialog: boolean;
}

export class Actions extends Component<ActionsProps, ActionsState> {
  constructor(props: ActionsProps) {
    super(props, { showEditDialog: props.showEditDialog ? true : false });
    this.hideDialog = this.hideDialog.bind(this);
    this.showDialog = this.showDialog.bind(this);
  }

  render() {
    return (
      <div id={this.props.id} className={this.className("Actions")}>
        {this.renderActions()}
        {this.renderEditButton()}
        {this.renderEditor()}
      </div>
    );
  }

  renderActions() {
    let actions: any[] = [];
    let a = this.props.txt.actions as any;
    for (let name in a) {
      let id = this.makeId(name);
      let action = a[name];
      if (action.isEnabled) {
        actions.push(
          <div
            key={id}
            id={id}
            className="Action"
            onClick={() => {
              this.props.onAction(action);
            }}
          >
            {action.name}
          </div>
        );
      }
    }

    return actions;
  }

  renderEditButton() {
    if (this.props.txt.isEditor) {
      return (
        <button
          onClick={this.showDialog}
          className="EditorButton"
          id={this.makeId("editorButton")}
        >
          <span>‚úè</span>
        </button>
      );
    } else {
      return undefined;
    }
  }

  renderEditor() {
    if (this.state.showEditDialog) {
      return (
        <ModalDialog
          id={this.makeId("dialog")}
          title="Actions"
          onCancel={this.hideDialog}
          onOkay={this.hideDialog}
        >
          <PropertyGrid
            id={this.makeId("propertyGrid")}
            object={this.props.txt.actions}
            onChange={(object: TxtActions) => {
              this.props.onChange(object);
            }}
          />
        </ModalDialog>
      );
    }
  }

  hideDialog() {
    this.setState((state: ActionsState) => {
      state.showEditDialog = false;
      return state;
    });
  }

  showDialog() {
    this.setState((state: ActionsState) => {
      state.showEditDialog = true;
      return state;
    });
  }
}

export default Actions;
