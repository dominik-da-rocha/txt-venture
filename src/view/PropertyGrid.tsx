import "./PropertyGrid.css";
import React, { ChangeEvent } from "react";
import Component, { ComponentProps, ComponentState } from "./Component";

export interface PropertyGridProps extends ComponentProps {
  object: any;
  onChange(object: any): void;
}

export interface PropertyGridState extends ComponentState {
  object: any;
  items: any;
}

export class PropertyGrid extends Component<
  PropertyGridProps,
  PropertyGridState
> {
  key: number = 0;

  constructor(props: PropertyGridProps) {
    super(props, {
      object: props.object,
      items: {},
    });
  }

  render() {
    this.key = 0;
    return (
      <div id={this.props.id} className={this.className("PropertyGrid")}>
        {this.renderRows(this.state.object, "object")}
      </div>
    );
  }

  renderRows(object: any, path: string): React.ReactNode {
    let items: any[] = [];
    for (let name in object) {
      if (!name.endsWith("_")) {
        let fullName = path + "." + name;
        let type = typeof object[name] === "object" ? "Object" : "Value";
        let key = "item-" + ++this.key;
        let id = this.makeId(key);
        items.push(
          <div key={key} className={type + " ExpanderContent"}>
            {this.renderExpander(type, id, key)}
            <label className="ExpanderLabel" htmlFor={id}>
              {name}
            </label>
            {this.renderValue(object[name], id, fullName)}
          </div>
        );
      }
    }
    return items;
  }

  renderExpander(type: string, id: string, key: string) {
    return type === "Object" ? (
      <input
        className="Expander"
        type="checkbox"
        id={id}
        defaultChecked={this.state.items[key] ? false : true}
        onChange={(event: ChangeEvent<HTMLInputElement>) => {
          let value = !event.target.checked;
          this.setState((state: PropertyGridState) => {
            state.items[key] = value;
            return state;
          });
        }}
      />
    ) : undefined;
  }

  renderValue(object: any, id: string, fullName: string): React.ReactNode {
    switch (typeof object) {
      case "boolean":
        return (
          <input
            id={id}
            className="Editor"
            type="checkbox"
            defaultChecked={object}
            onChange={(event: ChangeEvent<HTMLInputElement>) => {
              this.setObject(fullName, event.target.checked);
            }}
          />
        );
      case "string":
        return (
          <input
            id={id}
            className="Editor"
            type="text"
            defaultValue={object}
            onChange={(event: ChangeEvent<HTMLInputElement>) => {
              this.setObject(fullName, event.target.value);
            }}
          />
        );
      case "number":
        return (
          <input
            id={id}
            className="Editor"
            type="number"
            defaultValue={object.toString()}
            onChange={(event: ChangeEvent<HTMLInputElement>) => {
              this.setObject(fullName, event.target.valueAsNumber);
            }}
          />
        );
      case "object":
        return this.renderRows(object, fullName);
    }
  }
  
  setObject(fullName: string, value: string | boolean | number) {
    this.setState((state: PropertyGridState) => {
      let object: any = state;
      let path = fullName.split(".");
      path.forEach((name: string) => {
        if (name === path[path.length - 1]) {
          object[name] = value;
        } else {
          object = object[name];
        }
      });
      this.props.onChange(state.object);
      return state;
    });
  }
}

export default PropertyGrid;