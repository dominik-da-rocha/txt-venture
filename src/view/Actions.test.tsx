import Enzyme, { mount, render, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Renderer from "react-test-renderer";
import React from "react";
import Actions from "./Actions";
import TxtVenture, { TxtAction, TxtActions } from "../model/TxtVenture";
import PropertyGrid, { PropertyGridProps } from "./PropertyGrid";

Enzyme.configure({ adapter: new Adapter() });

describe("Actions", () => {
  it("renders without crashing", () => {
    expect(() => {
      shallow(
        <Actions
          id="test"
          txt={new TxtVenture()}
          onAction={(action: TxtAction) => {}}
          onChange={(actions: TxtActions) => {}}
        />
      );
    }).not.toThrow();
  });

  it("renders snapshot", () => {
    const tree = Renderer.create(
      <Actions
        id="test"
        className="Test"
        key="test"
        txt={new TxtVenture()}
        onAction={(action: TxtAction) => {}}
        onChange={(actions: TxtActions) => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders disabled action snapshot", () => {
    let txt = new TxtVenture();
    txt.actions.close.isEnabled = false;
    const tree = Renderer.create(
      <Actions
        id="test"
        className="Test"
        key="test"
        txt={txt}
        onAction={(action: TxtAction) => {}}
        onChange={(actions: TxtActions) => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders disabled editor snapshot", () => {
    let txt = new TxtVenture();
    txt.isEditor = false;
    const tree = Renderer.create(
      <Actions
        id="test"
        className="Test"
        key="test"
        txt={txt}
        onAction={(action: TxtAction) => {}}
        onChange={(actions: TxtActions) => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders editor snapshot", () => {
    let txt = new TxtVenture();
    txt.isEditor = false;
    const tree = Renderer.create(
      <Actions
        id="test"
        className="Test"
        key="test"
        txt={txt}
        showEditDialog={true}
        onAction={(action: TxtAction) => {}}
        onChange={(actions: TxtActions) => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("click action", () => {
    let testAction: TxtAction = {
      isEnabled: false,
      name: "",
    };

    let wrapper = shallow<Actions>(
      <Actions
        id="test"
        txt={new TxtVenture()}
        onAction={(action: TxtAction) => {
          testAction = action;
        }}
        onChange={(actions: TxtActions) => {}}
      />
    );
    wrapper.find("#test-talkTo").simulate("click");
    expect(testAction.name).toBe("Rede mit");
  });

  it("click editor", () => {
    let wrapper = shallow<Actions>(
      <Actions
        id="test"
        txt={new TxtVenture()}
        onAction={(action: TxtAction) => {}}
        onChange={(actions: TxtActions) => {}}
      />
    );
    expect(wrapper.state().showEditDialog).toBeFalsy();
    wrapper.find("#test-editorButton").simulate("click");
    expect(wrapper.state().showEditDialog).toBeTruthy();
  });

  it("click change from property grid", () => {
    let txt = new TxtVenture();
    let wrapper = mount<Actions>(
      <Actions
        id="test"
        txt={txt}
        showEditDialog={true}
        onAction={(action: TxtAction) => {}}
        onChange={(actions: TxtActions) => {
          txt.actions = actions;
        }}
      />
    );
    expect(txt.actions.close.name).toBe("SchlieÃŸe");
    wrapper
      .find("#test-propertyGrid-item-8")
      .simulate("change", { target: { value: "Hallo Welt" } });
    expect(txt.actions.close.name).toBe("Hallo Welt");
  });

  it("click hide dialoge", () => {
    let txt = new TxtVenture();
    let wrapper = mount<Actions>(
      <Actions
        id="test"
        txt={txt}
        showEditDialog={true}
        onAction={(action: TxtAction) => {}}
        onChange={(actions: TxtActions) => {
          txt.actions = actions;
        }}
      />
    );
    expect(wrapper.state().showEditDialog).toBeTruthy();
    wrapper.find("#test-dialog-x").simulate("click");
    expect(wrapper.state().showEditDialog).toBeFalsy();
  });
});
