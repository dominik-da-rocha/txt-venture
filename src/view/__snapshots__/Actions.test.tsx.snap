// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions renders disabled action snapshot 1`] = `
<div
  className="Component Actions Test"
  id="test"
>
  <div
    className="Action"
    id="test-giveTo"
    onClick={[Function]}
  >
    Gib
  </div>
  <div
    className="Action"
    id="test-open"
    onClick={[Function]}
  >
    Öffne
  </div>
  <div
    className="Action"
    id="test-pickUp"
    onClick={[Function]}
  >
    Nimm
  </div>
  <div
    className="Action"
    id="test-lookAt"
    onClick={[Function]}
  >
    Schau
  </div>
  <div
    className="Action"
    id="test-talkTo"
    onClick={[Function]}
  >
    Rede mit
  </div>
  <div
    className="Action"
    id="test-push"
    onClick={[Function]}
  >
    Drücke
  </div>
  <div
    className="Action"
    id="test-pull"
    onClick={[Function]}
  >
    Ziehe
  </div>
  <div
    className="Action"
    id="test-goto"
    onClick={[Function]}
  >
    Gehe zu
  </div>
  <div
    className="Action"
    id="test-use"
    onClick={[Function]}
  >
    Benutze
  </div>
  <div
    className="Action"
    id="test-save"
    onClick={[Function]}
  >
    Speichern
  </div>
  <div
    className="Action"
    id="test-load"
    onClick={[Function]}
  >
    Laden
  </div>
  <button
    className="EditorButton"
    id="test-editorButton"
    onClick={[Function]}
  >
    <span>
      ✏
    </span>
  </button>
</div>
`;

exports[`Actions renders disabled editor snapshot 1`] = `
<div
  className="Component Actions Test"
  id="test"
>
  <div
    className="Action"
    id="test-giveTo"
    onClick={[Function]}
  >
    Gib
  </div>
  <div
    className="Action"
    id="test-open"
    onClick={[Function]}
  >
    Öffne
  </div>
  <div
    className="Action"
    id="test-close"
    onClick={[Function]}
  >
    Schließe
  </div>
  <div
    className="Action"
    id="test-pickUp"
    onClick={[Function]}
  >
    Nimm
  </div>
  <div
    className="Action"
    id="test-lookAt"
    onClick={[Function]}
  >
    Schau
  </div>
  <div
    className="Action"
    id="test-talkTo"
    onClick={[Function]}
  >
    Rede mit
  </div>
  <div
    className="Action"
    id="test-push"
    onClick={[Function]}
  >
    Drücke
  </div>
  <div
    className="Action"
    id="test-pull"
    onClick={[Function]}
  >
    Ziehe
  </div>
  <div
    className="Action"
    id="test-goto"
    onClick={[Function]}
  >
    Gehe zu
  </div>
  <div
    className="Action"
    id="test-use"
    onClick={[Function]}
  >
    Benutze
  </div>
  <div
    className="Action"
    id="test-save"
    onClick={[Function]}
  >
    Speichern
  </div>
  <div
    className="Action"
    id="test-load"
    onClick={[Function]}
  >
    Laden
  </div>
</div>
`;

exports[`Actions renders editor snapshot 1`] = `
<div
  className="Component Actions Test"
  id="test"
>
  <div
    className="Action"
    id="test-giveTo"
    onClick={[Function]}
  >
    Gib
  </div>
  <div
    className="Action"
    id="test-open"
    onClick={[Function]}
  >
    Öffne
  </div>
  <div
    className="Action"
    id="test-close"
    onClick={[Function]}
  >
    Schließe
  </div>
  <div
    className="Action"
    id="test-pickUp"
    onClick={[Function]}
  >
    Nimm
  </div>
  <div
    className="Action"
    id="test-lookAt"
    onClick={[Function]}
  >
    Schau
  </div>
  <div
    className="Action"
    id="test-talkTo"
    onClick={[Function]}
  >
    Rede mit
  </div>
  <div
    className="Action"
    id="test-push"
    onClick={[Function]}
  >
    Drücke
  </div>
  <div
    className="Action"
    id="test-pull"
    onClick={[Function]}
  >
    Ziehe
  </div>
  <div
    className="Action"
    id="test-goto"
    onClick={[Function]}
  >
    Gehe zu
  </div>
  <div
    className="Action"
    id="test-use"
    onClick={[Function]}
  >
    Benutze
  </div>
  <div
    className="Action"
    id="test-save"
    onClick={[Function]}
  >
    Speichern
  </div>
  <div
    className="Action"
    id="test-load"
    onClick={[Function]}
  >
    Laden
  </div>
  <div
    className="Component ModalDialog"
    id="test-dialog"
  >
    <div
      className="Dialog"
    >
      <div
        className="Header"
      >
        <span>
          Actions
        </span>
        <span
          className="X"
          id="test-dialog-x"
          onClick={[Function]}
        >
          X
        </span>
      </div>
      <div
        className="Content"
      >
        <div
          className="Component PropertyGrid"
          id="test-propertyGrid"
        >
          <div
            className="Object ExpanderContent"
          >
            <input
              className="Expander"
              defaultChecked={true}
              id="test-propertyGrid-item-1"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ExpanderLabel"
              htmlFor="test-propertyGrid-item-1"
            >
              giveTo
            </label>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-2"
              >
                name
              </label>
              <input
                className="Editor"
                defaultValue="Gib"
                id="test-propertyGrid-item-2"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-3"
              >
                isEnabled
              </label>
              <input
                className="Editor"
                defaultChecked={true}
                id="test-propertyGrid-item-3"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
          <div
            className="Object ExpanderContent"
          >
            <input
              className="Expander"
              defaultChecked={true}
              id="test-propertyGrid-item-4"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ExpanderLabel"
              htmlFor="test-propertyGrid-item-4"
            >
              open
            </label>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-5"
              >
                name
              </label>
              <input
                className="Editor"
                defaultValue="Öffne"
                id="test-propertyGrid-item-5"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-6"
              >
                isEnabled
              </label>
              <input
                className="Editor"
                defaultChecked={true}
                id="test-propertyGrid-item-6"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
          <div
            className="Object ExpanderContent"
          >
            <input
              className="Expander"
              defaultChecked={true}
              id="test-propertyGrid-item-7"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ExpanderLabel"
              htmlFor="test-propertyGrid-item-7"
            >
              close
            </label>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-8"
              >
                name
              </label>
              <input
                className="Editor"
                defaultValue="Schließe"
                id="test-propertyGrid-item-8"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-9"
              >
                isEnabled
              </label>
              <input
                className="Editor"
                defaultChecked={true}
                id="test-propertyGrid-item-9"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
          <div
            className="Object ExpanderContent"
          >
            <input
              className="Expander"
              defaultChecked={true}
              id="test-propertyGrid-item-10"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ExpanderLabel"
              htmlFor="test-propertyGrid-item-10"
            >
              pickUp
            </label>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-11"
              >
                name
              </label>
              <input
                className="Editor"
                defaultValue="Nimm"
                id="test-propertyGrid-item-11"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-12"
              >
                isEnabled
              </label>
              <input
                className="Editor"
                defaultChecked={true}
                id="test-propertyGrid-item-12"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
          <div
            className="Object ExpanderContent"
          >
            <input
              className="Expander"
              defaultChecked={true}
              id="test-propertyGrid-item-13"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ExpanderLabel"
              htmlFor="test-propertyGrid-item-13"
            >
              lookAt
            </label>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-14"
              >
                name
              </label>
              <input
                className="Editor"
                defaultValue="Schau"
                id="test-propertyGrid-item-14"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-15"
              >
                isEnabled
              </label>
              <input
                className="Editor"
                defaultChecked={true}
                id="test-propertyGrid-item-15"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
          <div
            className="Object ExpanderContent"
          >
            <input
              className="Expander"
              defaultChecked={true}
              id="test-propertyGrid-item-16"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ExpanderLabel"
              htmlFor="test-propertyGrid-item-16"
            >
              talkTo
            </label>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-17"
              >
                name
              </label>
              <input
                className="Editor"
                defaultValue="Rede mit"
                id="test-propertyGrid-item-17"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-18"
              >
                isEnabled
              </label>
              <input
                className="Editor"
                defaultChecked={true}
                id="test-propertyGrid-item-18"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
          <div
            className="Object ExpanderContent"
          >
            <input
              className="Expander"
              defaultChecked={true}
              id="test-propertyGrid-item-19"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ExpanderLabel"
              htmlFor="test-propertyGrid-item-19"
            >
              push
            </label>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-20"
              >
                name
              </label>
              <input
                className="Editor"
                defaultValue="Drücke"
                id="test-propertyGrid-item-20"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-21"
              >
                isEnabled
              </label>
              <input
                className="Editor"
                defaultChecked={true}
                id="test-propertyGrid-item-21"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
          <div
            className="Object ExpanderContent"
          >
            <input
              className="Expander"
              defaultChecked={true}
              id="test-propertyGrid-item-22"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ExpanderLabel"
              htmlFor="test-propertyGrid-item-22"
            >
              pull
            </label>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-23"
              >
                name
              </label>
              <input
                className="Editor"
                defaultValue="Ziehe"
                id="test-propertyGrid-item-23"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-24"
              >
                isEnabled
              </label>
              <input
                className="Editor"
                defaultChecked={true}
                id="test-propertyGrid-item-24"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
          <div
            className="Object ExpanderContent"
          >
            <input
              className="Expander"
              defaultChecked={true}
              id="test-propertyGrid-item-25"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ExpanderLabel"
              htmlFor="test-propertyGrid-item-25"
            >
              goto
            </label>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-26"
              >
                name
              </label>
              <input
                className="Editor"
                defaultValue="Gehe zu"
                id="test-propertyGrid-item-26"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-27"
              >
                isEnabled
              </label>
              <input
                className="Editor"
                defaultChecked={true}
                id="test-propertyGrid-item-27"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
          <div
            className="Object ExpanderContent"
          >
            <input
              className="Expander"
              defaultChecked={true}
              id="test-propertyGrid-item-28"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ExpanderLabel"
              htmlFor="test-propertyGrid-item-28"
            >
              use
            </label>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-29"
              >
                name
              </label>
              <input
                className="Editor"
                defaultValue="Benutze"
                id="test-propertyGrid-item-29"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-30"
              >
                isEnabled
              </label>
              <input
                className="Editor"
                defaultChecked={true}
                id="test-propertyGrid-item-30"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
          <div
            className="Object ExpanderContent"
          >
            <input
              className="Expander"
              defaultChecked={true}
              id="test-propertyGrid-item-31"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ExpanderLabel"
              htmlFor="test-propertyGrid-item-31"
            >
              save
            </label>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-32"
              >
                name
              </label>
              <input
                className="Editor"
                defaultValue="Speichern"
                id="test-propertyGrid-item-32"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-33"
              >
                isEnabled
              </label>
              <input
                className="Editor"
                defaultChecked={true}
                id="test-propertyGrid-item-33"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
          <div
            className="Object ExpanderContent"
          >
            <input
              className="Expander"
              defaultChecked={true}
              id="test-propertyGrid-item-34"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ExpanderLabel"
              htmlFor="test-propertyGrid-item-34"
            >
              load
            </label>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-35"
              >
                name
              </label>
              <input
                className="Editor"
                defaultValue="Laden"
                id="test-propertyGrid-item-35"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="Value ExpanderContent"
            >
              <label
                className="ExpanderLabel"
                htmlFor="test-propertyGrid-item-36"
              >
                isEnabled
              </label>
              <input
                className="Editor"
                defaultChecked={true}
                id="test-propertyGrid-item-36"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="Footer"
      >
        <button
          id="test-dialog-okay"
          onClick={[Function]}
        >
          Okay
        </button>
        <button
          id="test-dialog-cancel"
          onClick={[Function]}
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Actions renders snapshot 1`] = `
<div
  className="Component Actions Test"
  id="test"
>
  <div
    className="Action"
    id="test-giveTo"
    onClick={[Function]}
  >
    Gib
  </div>
  <div
    className="Action"
    id="test-open"
    onClick={[Function]}
  >
    Öffne
  </div>
  <div
    className="Action"
    id="test-close"
    onClick={[Function]}
  >
    Schließe
  </div>
  <div
    className="Action"
    id="test-pickUp"
    onClick={[Function]}
  >
    Nimm
  </div>
  <div
    className="Action"
    id="test-lookAt"
    onClick={[Function]}
  >
    Schau
  </div>
  <div
    className="Action"
    id="test-talkTo"
    onClick={[Function]}
  >
    Rede mit
  </div>
  <div
    className="Action"
    id="test-push"
    onClick={[Function]}
  >
    Drücke
  </div>
  <div
    className="Action"
    id="test-pull"
    onClick={[Function]}
  >
    Ziehe
  </div>
  <div
    className="Action"
    id="test-goto"
    onClick={[Function]}
  >
    Gehe zu
  </div>
  <div
    className="Action"
    id="test-use"
    onClick={[Function]}
  >
    Benutze
  </div>
  <div
    className="Action"
    id="test-save"
    onClick={[Function]}
  >
    Speichern
  </div>
  <div
    className="Action"
    id="test-load"
    onClick={[Function]}
  >
    Laden
  </div>
  <button
    className="EditorButton"
    id="test-editorButton"
    onClick={[Function]}
  >
    <span>
      ✏
    </span>
  </button>
</div>
`;
